pkg/nsx/services/subnet/builder.go

```diff
@@ -23,26 +25,37 @@ func getCluster(service *SubnetService) string {
 	return service.NSXConfig.Cluster
 }
 
-func (service *SubnetService) BuildSubnetID(subnet *v1alpha1.Subnet) string {
-	return util.GenerateIDByObject(subnet)
+// BuildSubnetID uses format "subnet.Name_$(hash(${subnet.UUID}))[5]" to generate the VpcSubnet's id.
```

Comment:
Just to confirm, is the format actually `subnet.Name_$(hash(${namespace.UUID}))[5]` instead of `subnet.Name_$(hash(${subnet.UUID}))[5]`?

Reply:
Yes, it is following the discussion https://chat.google.com/room/AAAAQU8Mo9Q/PzKGouruGi8/F-9ndThvzKM?cls=10 which suggested to use Namespace UUID if possible. Now we only use CR's UUID in securitypoilcy which has more complicated use cases, other resources are actually using Namespace's UUID. It is expected to reduce the collision from UUIDs. A corner case is we may still hit the collisions if two CRs name's prefix are the same in the same Namespace, we may use a random UUID to generate a different hash in this case.